Plutarch/ByteString.hs

```diff
@@ -37,6 +47,33 @@ phexByteStr = punsafeConstant . PLC.Some . PLC.ValueOf PLC.DefaultUniByteString
 pbyteStr :: ByteString -> Term s PByteString
 pbyteStr = punsafeConstant . PLC.Some . PLC.ValueOf PLC.DefaultUniByteString
 
+-----------------------------------------------------------
+-- The following functions should be import qualified. --
+-----------------------------------------------------------
+
+-- | Prepend a byte, represented by a non negative 'PInteger', to a 'PBytestring'.
+pcons :: Term s (PInteger :--> PByteString :--> PByteString)
+pcons = punsafeBuiltin PLC.ConsByteString
+
+{- | Slice a 'PByteString' with given start and end indices.
+
+>>> (pslice # 1 # 3 phexByteStr "4102afde5b2a") #== phexByteStr "02afde"
+-}
+pslice :: Term s (PInteger :--> PInteger :--> PByteString :--> PByteString)
+pslice = punsafeBuiltin PLC.SliceByteString
+
+-- | Find the length of a 'PByteString'.
+plength :: Term s (PByteString :--> PInteger)
+plength = punsafeBuiltin PLC.LengthOfByteString
+
+-- | 'PByteString' indexing function.
+pindex :: Term s (PByteString :--> PInteger :--> PInteger)
+pindex = punsafeBuiltin PLC.IndexByteString
```

Comment:
I don't think the names should be generic, though I realise that this is how `Data.ByteString` does it.

Reply:
I spent too long debating what I should've done here. In one of the commits in this PR, these were not generic names. I made them generic after the fact thinking "hey that's how haskell containers do it".

I honestly don't know which choice is better here - going to go with whatever you (or others) suggest.

Reply:
The issue IMO is that we haven't done this until now. E.g. `pheadBuiltin`. I think we should do the same here and write `pbindex` or something to make it fit with the existing naming convention.

Reply:
You know what, it doesn't matter.

Reply:
How about these names-
https://github.com/Plutonomicon/plutarch/pull/30/commits/da57217b6a12fbe0f6ee87a2c4950169f6669416

Should these go back to those names?

Reply:
I think we should still choose a name consistent with everything else though. E.g. `pindexBS`, to make it similar to `pheadBuiltin` and such.

Reply:
Sure, done!